cmd/promoted-image-governor/main.go

```diff
@@ -270,6 +281,67 @@ func isMirroredFromOCP(tag api.ImageStreamTagReference, refs []ImageStreamRef) b
 	return false
 }
 
+type refGetter interface {
+	GetRef(org, repo, ref string) (string, error)
+}
+
+type gitRefGetter struct {
+	clone func(org, repo string) (git.RepoClient, error)
+}
+
+func (g *gitRefGetter) GetRef(org, repo, ref string) (ret string, retError error) {
+	r, err := g.clone(org, repo)
```

Comment:
@stevekuznetsov 
It will git-clone all repos if it builds promoted images, right?
Seems it is lots of work.

Reply:
Or you mean this?

```
$ git ls-remote https://github.com/openshift/images.git release-4.0
2f60da39a9d2e5cc00293b8ec7ad559fcd32446a	refs/heads/release-4.0
```

Reply:
`ls-remote` is the more efficient way, yes - we should already be doing this in promotion reconciler and have utilities for it upstream.

Reply:
promotion reconciler uses github client (that is where my first draft came from).
https://github.com/openshift/ci-tools/blob/73781b2ee20eee321a86d1de080e8b879d882737/pkg/controller/promotionreconciler/reconciler.go#L212-L215

We could switch that later if this one goes well.

I cannot find the API upstream.
The closest thing is 
https://github.com/kubernetes/test-infra/blob/2e2328ea3fa8bd91b11122b848b539a74292cb60/prow/git/v2/interactor.go#L151